posthog/warehouse/models/snapshot_job.py

```diff
@@ -0,0 +1,21 @@
+from django.db import models
+
+from posthog.models.team import Team
+from posthog.models.utils import CreatedMetaFields, DeletedMetaFields, UUIDTModel
+
+
+class DataWarehouseSnapshotJob(CreatedMetaFields, UUIDTModel, DeletedMetaFields):
+    class Status(models.TextChoices):
+        RUNNING = "Running", "Running"
+        COMPLETED = "Completed", "Completed"
+        FAILED = "Failed", "Failed"
+
+    team = models.ForeignKey(Team, on_delete=models.CASCADE)
+    config = models.ForeignKey("posthog.DataWarehouseSnapshotConfig", on_delete=models.CASCADE)
+
+    # Each snapshot job will create a saved query that is filtered to the state of the table at run time
+    table = models.ForeignKey("posthog.DataWarehouseSavedQuery", on_delete=models.CASCADE, null=True, blank=True)
```

Comment:
`table` - or should we call this `saved_query` to match the rest of our naming standards